File: 2017/day_5/quiz_1.rb
input = File.readlines('input.txt').map{ |line| line.strip.to_i }
range = (0..input.size - 1)
cursor = 0
steps = 0

while range.include?(cursor)
  move = input[cursor]
  input[cursor] += 1
  cursor += move

  steps+=1
end

p steps

File: 2024/day_1/quiz_1.py
def read_input(filename):
  try:
    with open(filename, 'r') as file:
      return file.readlines()
  except FileNotFoundError:
    print(f'File {filename} not found')
  except Exception as e:
    print(f'An error occurred: {e}')

if __name__ == '__main__':
  input_data = read_input('input.txt')
  left_list = []
  right_list = []
  for line in input_data:
    left, right = map(int, line.strip().split())
    left_list.append(left)
    right_list.append(right)

  left_list.sort()
  right_list.sort()
  diff_array = [abs(a -b) for a, b in zip(left_list, right_list)]

  sum_diff = sum(diff_array)
  print(sum_diff)
